### DISCUSSION TITLE
How/where do you get the data for generating these AstroCharts?

### DISCUSSION THREAD

**Discussion by lancejpollard** [AstroDraw/AstroChart#79] [Q&A] [answered]
Where do you get the data used to render the charts, is there an open source repo for computing that somewhere, or how does it work generally? I am new to this, thanks for the help!

You show this example:

```js
//data example
var data = {
  "planets":{
    "Pluto":[63], 
    "Neptune":[110], 
    "Uranus":[318], 
    "Saturn":[201], 
    "Jupiter":[192], 
    "Mars":[210], 
    "Moon":[268], 
    "Sun":[281],
    "Mercury":[312], 
    "Venus":[330]},
  "cusps":[296, 350, 30, 56, 75, 94, 116, 170, 210, 236, 255, 274]          
};

var radix = chart.radix( data );
```

But where do I get that data from? How do I compute/generate it?

**Comment by afucher** [comment, marked as answer]
Hey @lancejpollard , people usually uses the Swiss Ephemeris from [astro.com](https://astro.com)
https://github.com/aloistr/swisseph

**Reply by lancejpollard** [reply]
Thank you!
